library/SimplePie/Enclosure.php

```diff
@@ -1152,6 +1152,11 @@ public function get_real_type($find_handler = false)
 		// If we encounter an unsupported mime-type, check the file extension and guess intelligently.
 		if (!in_array($type, array_merge($types_flash, $types_fmedia, $types_quicktime, $types_wmedia, $types_mp3)))
 		{
+			$extension = $this->get_extension();
+			if ($extension === null) {
+				return null;
+			}
+
 			switch (strtolower($this->get_extension()))
```

Comment:
could you change this line to use $extension?

Reply:
Of course. Done.